packages/curvenote-cli/src/submissions/submit.utils.ts

```diff
@@ -61,41 +87,123 @@ export function venueQuestion(session: ISession) {
-  try {
-    kinds = await getFromJournals(session, `sites/${venue}/kinds`);
-  } catch (err: any) {
+  const openCollections = collections.items.filter((c) => c.open);
+  if (openCollections.length === 0) {
+    session.log.info(
+      `${chalk.red(`⛔️ no collections are open for submissions at venue "${venue}"`)}`,
+    );
+    process.exit(1);
+  }
+
+  session.log.debug(`Explicit collection provided: ${opts?.collection}`);
+  let selectedCollection = opts?.collection
+    ? openCollections.find(
+        (c) => c.slug === (opts?.collection === 'default' ? '' : opts?.collection),
+      )
+    : undefined;
+
+  if (opts?.collection && (!selectedCollection || !selectedCollection?.open)) {
+    session.log.info(
+      `${chalk.red(`⛔️ collection "${opts?.collection}" is not open for submissions at venue "${venue}"`)}`,
+    );
     session.log.info(
-      `${chalk.red(`🚨 could not get submission kinds listing from venue ${venue}`)}`,
+      `${chalk.bold(`🗂 open collections are: ${openCollections.map((c) => collectionMoniker(c)).join(', ')}`)}`,
     );
     process.exit(1);
   }
 
-  let kind;
+  if (!selectedCollection && openCollections.length === 1) {
+    selectedCollection = openCollections[0];
+  } else if (!selectedCollection) {
+    const response = await inquirer.prompt([collectionQuestions(venue, openCollections)]);
+    selectedCollection = response.collections;
+  }
+
+  if (!selectedCollection) {
+    session.log.info(`${chalk.red(`⛔️ could not determine the collection to submit to`)}`);
+    process.exit(1);
+  }
+  session.log.info(`🗂  Collection "${collectionMoniker(selectedCollection)}" selected`);
+
+  const kind = await determineSubmissionKindFromCollection(
+    session,
+    venue,
+    selectedCollection,
+    opts,
+  );
+
+  return { collection: selectedCollection, kind };
+}
+
+export async function getSubmissionKind(
+  session: ISession,
+  venue: string,
+  kindIdOrName: string,
+): Promise<SubmissionKindDTO> {
+  const kind = await getFromJournals(session, `sites/${venue}/kinds/${kindIdOrName}`);
+  if (!kind) throw new Error('kind not found');
+  return kind;
+}
+
+export async function listSubmissionKinds(
+  session: ISession,
+  venue: string,
+): Promise<SubmissionKindsDTO> {
+  return getFromJournals(session, `sites/${venue}/kinds`);
+}
+
+export async function determineSubmissionKindFromCollection(
+  session: ISession,
+  venue: string,
+  collection: CollectionDTO,
+  opts?: { kind?: string },
+) {
+  const kinds = collection.kinds;
+
+  let kindId;
   if (opts?.kind) {
-    const match = kinds.items.find(
+    const match = kinds.find(
       ({ name }: { name: string }) => name.toLowerCase() === opts.kind?.toLowerCase(),
     );
     if (!match) {
       session.log.info(
-        `${chalk.red(`🚨 submission kind "${opts.kind}" is not accepted at venue ${venue}`)}`,
+        `${chalk.bold.red(`⛔️ submission kind "${opts.kind}" is not accepted in the collection "${collectionMoniker(collection)}"`)}`,
```

Comment:
At some point we will probably want to massage this logging so it covers the UX for both (1) site with collections and (2) site with only a default collection but no collections as far as the user is concerned.

I think less nuance is ok for now though...

Reply:
yeah the messaging UX overall probably could be improved